integration_tests/cork_test.go

```diff
@@ -22,292 +20,456 @@ import (
+
+		// make it rain
+		for i := range s.chain.stewards {
+			addr := fmt.Sprintf("localhost:%v", STEWARD_PORT+i)
+			conn, err := grpc.Dial(
+				addr,
+				grpc.WithBlock(),
+				grpc.WithTransportCredentials(creds),
+			)
+			s.Require().NoError(err)
+			defer conn.Close()
 
-			return false
-		}, 1*time.Minute, 5*time.Second, "cellar event never seen")
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
 
-		s.T().Logf("waiting for gravity to submit call to cellar")
-		s.Require().Eventuallyf(func() bool {
-			s.T().Log("querying gravity logic call events...")
-			ethClient, err := ethclient.Dial(fmt.Sprintf("http://%s", s.ethResource.GetHostPort("8545/tcp")))
+			c := steward_proto.NewContractCallClient(conn)
+			s.T().Logf("sending request to %s", s.chain.validators[i].keyInfo.GetAddress())
+			_, err = c.Submit(ctx, request)
 			s.Require().NoError(err)
+		}
 
-			// For non-anonymous events, the first log topic is a keccak256 hash of the
-			// event signature.
-			eventSignature := []byte("LogicCallEvent(bytes32,uint256,bytes,uint256)")
-			logicCallEventSignatureTopic := crypto.Keccak256Hash(eventSignature)
-			query := ethereum.FilterQuery{
-				FromBlock: nil,
-				ToBlock:   nil,
-				Addresses: []common.Address{
-					gravityContract,
-				},
-				Topics: [][]common.Hash{
-					{
-						logicCallEventSignatureTopic,
-					},
-				},
-			}
-			result, err := ethClient.FilterLogs(context.Background(), query)
-			s.Require().NoError(err)
-			ethClient.Close()
+		return true
+	}, 100*time.Second, 1*time.Second, "Cork request took too long")
+}
 
-			s.T().Logf("got %v LogicCallEvent logs", len(result))
-			if len(result) == 0 {
-				return false
-			}
+func (s *IntegrationTestSuite) queryLogicCallTransaction(invalidationScope []byte, invalidationNonce int) {
+	s.T().Log("querying gravity module for logic call transaction")
+	val := s.chain.validators[0]
+	kb, err := val.keyring()
+	s.Require().NoError(err)
+
+	clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kb, "val", val.keyInfo.GetAddress())
+	s.Require().NoError(err)
+
+	gravityQueryClient := gravityTypes.NewQueryClient(clientCtx)
+	s.Require().Eventuallyf(func() bool {
+		request := &gravityTypes.ContractCallTxRequest{
+			InvalidationNonce: uint64(invalidationNonce),
+			InvalidationScope: invalidationScope,
+		}
+		response, _ := gravityQueryClient.ContractCallTx(context.Background(), request)
+
+		if response != nil {
+			s.T().Logf("found logic call %v!", response.LogicCall)
+			return true
+		}
 
-			// only one LogicCallEvent is expected
-			s.Require().Equal(len(result), 1)
+		return false
+	}, 1*time.Minute, 5*time.Second, "cellar event never seen")
+	time.Sleep(time.Duration(2000000000))
+}
 
-			log := result[0]
-			gravity_abi, err := GravityMetaData.GetAbi()
-			s.Require().NoError(err)
+func (s *IntegrationTestSuite) waitForLogicCallEvent(topic common.Hash, invalidationScope []byte, invalidationNonce int) {
```

Comment:
Was this helper meant to be used? It looks like the ethereum querying is still done manually in each test.

Reply:
It is, renamed to make it clearer. This is specifically for the Gravity contract, the manual one if for events in the mock cellar.